components/CraftItemRequirementsListInput.tsx

```diff
@@ -0,0 +1,292 @@
+    () => !amount || Number.isNaN(Number(amount)) || amount === '0',
+    [amount],
+  );
+
+  const moreThanSupply = useMemo(() => {
+    if (BigInt(amount) > BigInt(itemSupply)) {
+      return true;
+    }
+    return false;
+  }, [amount, itemSupply]);
+
+  const moreThanDistribution = useMemo(() => {
+    if (BigInt(amount) > BigInt(itemDistribution)) {
+      return true;
+    }
+    return false;
+  }, [amount, itemDistribution]);
+
+  useEffect(() => {
+    setShowError(false);
+  }, [selectedItem, amount]);
+
+  const errorText = useMemo(() => {
+    if (!selectedItem && amountInvalid) {
+      return 'Item and amount are invalid';
+    }
+    if (!selectedItem && !amountInvalid) {
+      return 'Item is invalid';
+    }
+    if (selectedItem && amountInvalid) {
+      return 'Amount is invalid';
+    }
+    if (!!selectedItem && moreThanSupply) {
+      return 'Amount exceeds item supply';
+    }
+    if (!!selectedItem && moreThanDistribution) {
+      return 'Amount exceeds item distribution';
+    }
+    return '';
+  }, [selectedItem, amountInvalid, moreThanSupply, moreThanDistribution]);
+
+  const onAddCraftItemRequirement = useCallback(() => {
+    if (
+      !selectedItem ||
+      amountInvalid ||
+      moreThanSupply ||
+      moreThanDistribution
+    ) {
+      setShowError(true);
+      return;
+    }
+
+    setRequirementsList(oldList => [
+      ...oldList,
+      {
+        itemId: BigInt(selectedItem.itemId).toString(),
+        amount: BigInt(amount).toString(),
+      },
+    ]);
+    setSelectedItem(null);
+    setAmount('');
+    setShowError(false);
+  }, [
+    amount,
+    selectedItem,
+    amountInvalid,
+    moreThanSupply,
+    moreThanDistribution,
+    setRequirementsList,
+  ]);
+
+  return (
+    <VStack spacing={4} w="100%" mb={4}>
+      <FormControl>
+        <Grid
+          w="100%"
+          templateColumns={{
+            base: '1fr',
+            sm: '3fr 1.5fr',
+            md: '3fr 1fr',
+          }}
+          gridGap={4}
+          position="relative"
+          justifyContent="center"
+          alignItems="center"
+        >
+          <SelectItemInput
+            items={items.filter(i => !i.soulbound)}
+            selectedItem={selectedItem}
+            setSelectedItem={setSelectedItem}
+          />
+          <Tooltip label={items.length === 0 ? 'No items available' : ''}>
+            <Input
+              type="number"
+              step={1}
+              min={0}
+              placeholder="Amount"
+              value={amount}
+              onChange={e => setAmount(e.target.value)}
+              isInvalid={amountInvalid && showError}
+              isDisabled={items.length === 0}
+              h="2.7125rem"
+            />
+          </Tooltip>
+        </Grid>
+        {showError && (
+          <FormHelperText color="red.500">{errorText}</FormHelperText>
+        )}
+      </FormControl>
+      <Tooltip label={items.length === 0 ? 'No items available' : ''}>
```

Comment:
Same as above comment

Reply:
fixed in [846ae47](https://github.com/raid-guild/character-sheets-frontend/pull/170/commits/846ae471e0454be0c7b863960a53c5166866b938)